source/MaterialXRuntime/RtNode.cpp

```diff
@@ -137,6 +159,24 @@ size_t RtNode::numOutputs() const
     return prim()->numOutputs();
 }
 
+const RtTokenVec& RtNode::getInputPublicMetadata(const RtToken& name) const
```

Comment:
Maybe name this method getPortPublicMetadata() instead and let it account for both inputs and outputs, since outputs can also have metadata. Inputs and outputs cannot have the same name within a prim so getting the port by name should work fine, and then we can find out if it's an input or output.

I suppose this should also be a virtual method on the schema base class. And also overridden by RtNodeDef to return the metadata names of nodedef ports as well.

Reply:
What should the RtNodeDef port values be? The same as RtNode ones?

Reply:
It's listed in the spec, page 51-52. (http://www.materialx.org/assets/MaterialX.v1.38.Spec.pdf)